apps/frontend/src/pages/CreateAccount.tsx

```diff
@@ -1,5 +1,206 @@
+import React, { useState } from "react";
+import {
+  Box,
+  Button,
+  TextField,
+  Typography,
+  FormHelperText,
+  Link,
+} from "@mui/material";
+import martletPlaceLogo from "../images/martletplace-logo.png";
+import { colors } from "../styles/colors";
+import { useStyles } from "../styles/pageStyles";
+import { useNavigate } from "react-router-dom";
+import axios from "axios";
+
 const CreateAccount = () => {
-  return <h1>CreateAccount</h1>;
+  const classes = useStyles();
+  const [email, setEmail] = useState("");
+  const [name, setName] = useState("");
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+
+  const [emailError, setEmailError] = useState("");
+  const [usernameError, setUsernameError] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+
+  const isFormEmpty = !email || !name || !username || !password;
+
+  const navigate = useNavigate();
+
+  const handleCreateAccount = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    // Regex for UVic email validation
+    const emailFormat = /^[^@]+@uvic\.ca$/;
+
+    // Regex for username validation
+    const usernameFormat = /^[a-zA-Z]{1,20}$/;
+
+    // Regex for password validation
+    const passwordFormat =
+      /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[\W_])[A-Za-z\d\W_]{8,}$/;
+
+    if (!emailFormat.test(email)) {
+      // Add or update an error state for email validation
+      setEmailError("Email must be a valid UVic email address.");
+      return;
+    } else {
+      // Clear email error if valid
+      setEmailError("");
+    }
+
+    if (!usernameFormat.test(username)) {
+      // Add or update an error state for username validation
+      setUsernameError(
+        "Username must be between 1 and 20 characters and only contain letters.",
+      );
+      return;
+    } else {
+      // Clear username error if valid
+      setUsernameError("");
+    }
+
+    if (!passwordFormat.test(password)) {
```

Comment:
nit: that regex legitimately scares me. How about switching to a simple validation that humans can easily comprehend?

```suggestion
    if (password.length >= 8 && /[a-z]/.test(password) && /[A-Z]/.test(password) && /[0-9]/.test(password) && /[\W_]/.test(password)) {
```